android-components/components/browser/state/src/test/java/mozilla/components/browser/state/engine/middleware/EngineDelegateMiddlewareTest.kt

```diff
@@ -742,4 +745,67 @@ class EngineDelegateMiddlewareTest {
         verify(engineSession1).purgeHistory()
         verify(engineSession2).purgeHistory()
     }
+
+    @Test
+    fun `TranslateAction correctly sets progress state AND begins a translation`() {
+        val tab = createTab("https://www.mozilla.org")
+        val engineSession: EngineSession = mock()
+        val engine: Engine = mock()
+        doReturn(engineSession).`when`(engine).createSession()
+        val store = BrowserStore(
+            middleware = EngineMiddleware.create(
+                engine = engine,
+                scope = scope,
+            ),
+            initialState = BrowserState(
+                tabs = listOf(tab),
+            ),
+        )
+
+        assertEquals(false, store.state.findTab(tab.id)?.translationsState?.isTranslateProcessing)
```

Comment:
Use `assertFalse`. Same thing below

Reply:
Updated, is it okay to force unwrap in a test?

Reply:
Definitely. The test will just fail anyway if it crashes